pkg/controller/controller.go

```diff
@@ -919,3 +950,57 @@ func validateIsolationGroups(cluster *myspec.M3DBCluster) error {
 	}
 	return nil
 }
+
+// updatedStatefulSet checks if we should update the actual StatefulSet to match it's
+// expected state. The method returns the updated StatefulSet if so, otherwise nil.
+func updatedStatefulSet(
+	actual *appsv1.StatefulSet, cluster *myspec.M3DBCluster, isoGroup myspec.IsolationGroup,
+) (*appsv1.StatefulSet, error) {
```

Comment:
Can we modify the return signature to be `*StatefulSet, bool, error` and use the bool as whether we upgraded rather than implicit `nil`?